packages/rocketchat-livechat/server/api/lib/livechat.js

```diff
@@ -0,0 +1,98 @@
+import _ from 'underscore';
+import LivechatVisitors from '../../models/LivechatVisitors';
+
+export const online = RocketChat.models.Users.findOnlineAgents().count() > 0;
+
+export function findTriggers() {
+	return RocketChat.models.LivechatTrigger.findEnabled().fetch().map((trigger) => _.pick(trigger, '_id', 'actions', 'conditions'));
+}
+
+export function findDepartments() {
+	return RocketChat.models.LivechatDepartment.findEnabledWithAgents().fetch().map((department) => _.pick(department, '_id', 'name', 'showOnRegistration'));
+}
+
+export function findGuest(token) {
+	return LivechatVisitors.getVisitorByToken(token, {
+		fields: {
+			name: 1,
+			username: 1,
+			token: 1,
+		},
+	});
+}
+
+export function findRoom(token, rid) {
+	const fields = {
+		departmentId: 1,
+		servedBy: 1,
+		open: 1,
+	};
+
+	if (!rid) {
+		return RocketChat.models.Rooms.findLivechatByVisitorToken(token, fields);
+	}
+
+	return RocketChat.models.Rooms.findLivechatByIdAndVisitorToken(rid, token, fields);
+}
+
+export function getRoom(guest, rid) {
+	const token = guest && guest.token;
+
+	const message = {
+		_id: Random.id(),
+		rid,
+		msg: '',
+		token,
+		ts: new Date(),
+	};
+
+	return RocketChat.Livechat.getRoom(guest, message, { jitsiTimeout: new Date(Date.now() + 3600 * 1000) });
```

Comment:
the `jitsiTimeout` param should be used only on [video.call](https://github.com/RocketChat/Rocket.Chat/pull/11900/files#diff-cf473363d900cdee2f518545b5e41f2aR3) endpoint..